Avoid failing when git user details are not configured

OldManYellsAtCloud opened the pull request:
In case a git username and email are not configured, then git errors
out when committing the initial and later the changed configuration
of the setup. This can happen if bitbake-setup is running in a
container, or for example if the user just doesn't use git for
committing.

To allow the script to still finish, verify upon start if the user
has the necessary details (name and email) configured in git. If not,
then use dummy details for committing.

Unfortunately `--author=Bitbake Setup <noreply@yoctoproject.org>` 
git argument isn't sufficient, git seems to require `user.name` and
 `user.email` configuration items specifically.

Once I fixed it at the first place, it also failed at a second one, so rather 
changed all git commands.

OldManYellsAtCloud committed:
bitbake-setup: fix typo

Signed-off-by: Gyorgy Sarvari <skandigraun@gmail.com>

OldManYellsAtCloud committed:
bitbake-setup: fix non-interactive init instructions

The --non-interactive option is position-sensitive, it needs to stand
after the "init" command, otherwise argparser complains:

bitbake-setup: error: unrecognized arguments: --non-interactive

Signed-off-by: Gyorgy Sarvari <skandigraun@gmail.com>

kanavin commented:
Thanks for the patches. I'm in OSS Amsterdam for the next few days, and likely won't have time to look at it properly until it's over. But I'll get to it!

kanavin commented on bin/bitbake-setup:
```diff
@@ -391,8 +392,8 @@ def init_config(settings, args, d):
-    bb.process.run("git -C {} init -b main".format(confdir))
-    bb.process.run("git -C {} commit --allow-empty -m 'Initial commit'".format(confdir))
+    bb.process.run("{} -C {} init -b main".format(git_cmd, confdir))
+    bb.process.run("{} -C {} commit --allow-empty -m 'Initial commit'".format(git_cmd, confdir))
```
I think we can avoid all this extra complexity by inserting these two between 'init' and 'commit' above:
```
git config user.name bitbake-setup
git config user.email bitbake-setup@not.set
```
This would not affect the global config, and as commits are created automatically, they should come from bitbake-setup, not from the user running it.

OldManYellsAtCloud replied:
Sounds good to me, done.

OldManYellsAtCloud committed:
bitbake-setup: make sure git user is always configured

In case a git username and email are not configured, then git errors
out when commiting the initial and later the changed configuration
of the setup. This can happen if bitbake-setup is running in a
container, or for example if the user just doesn't use git for
committing.

To avoid this failure, configure a dummy user to be used with
comitting the config changes.

Signed-off-by: Gyorgy Sarvari <skandigraun@gmail.com>
